xc/xc7/techmap/fix_carry.py

```diff
@@ -0,0 +1,380 @@
+""" Script for addressing CARRY4 output congestion in elaborated netlists.
+
+Usage:
+
+    python3 fix_carry.py < input-netlist-json > output-netlist-json
+
+Description:
+
+    In the 7-series SLICEL (and SLICEM) sites, there can be output congestion
+    if both the CO and O of the CARRY4 are used.  This congestion can be
+    avoided if transparent/open latches or registers on the output of the
+    CARRY4.
+
+    VPR does not currently support either of those options, so for now, if
+    both CO and O are used, the CO output is converted into a LUT equation to
+    recompute the CO output from O, DI and S.  See carry_map.v and
+    clean_carry_map.v for details.
+
+    This script identifies CARRY4 chains in the netlist, identifies if there
+    is output congestion on the O and CO ports, and marks the congestion by
+    changing CARRY_CO_DIRECT (e.g. directly use the CO port) to CARRY_CO_LUT
+    (compute the CO value using a LUT equation).
```

Comment:
This would really benefit from some good diagrams.

Reply:
Diagrams added, take a look